src/polyswarm_api/types/resources.py

```diff
@@ -236,19 +242,21 @@ def __init__(self, handle, artifact_name=None, artifact_type=None, polyswarm=Non
         :param analyze: Boolean, if True will run analyses on artifact on startup (Note: this may still run later if False)
         """
         # create the LocalHandle with the given handle and don't write anything to it
-        super(LocalArtifact, self).__init__(b'', polyswarm=polyswarm, handle=handle)
-        self.artifact_type = artifact_type or ArtifactType.FILE
-        self.artifact_name = artifact_name or self.hash
-
         self.sha256 = None
         self.sha1 = None
         self.md5 = None
         self.analyzed = False
+
+        super(LocalArtifact, self).__init__(b'', polyswarm=polyswarm, handle=handle)
```

Comment:
We should call `super` before making changes specific to this instance. The parent class has "more generic" code than the "child class", and the execution of the child `__init__` should be executed after so that these are the changes that supersede the changes in the inherited `__init__`.

Reply:
Sure, just got too excited moving those variables around. `hash` wasn't resolvable until those attributes existed.